crucible-mir/src/Mir/Mir.hs

```diff
@@ -592,8 +592,6 @@ makeWrapped ''Substs
 instance Semigroup Collection where
   (Collection f1 a1 a1' t1 s1 v1 n1 tys1 r1) <> (Collection f2 a2 a2' t2 s2 v2 n2 tys2 r2) =
     Collection (f1 <> f2) (a1 <> a2) (a1' <> a2') (t1 <> t2) (s1 <> s2) (v1 <> v2) (n1 <> n2) (tys1 <> tys2) (r1 <> r2)
```

Comment:
Thinking about this some more, I think we should remove the `Semigroup` instances for `Collection` and friends as well. The reasoning is similar: when we add a schema version number to `Collection`, it's not clear how one would implement `(<>)` for two version numbers. (The only uses of `(<>)` in `crucible-mir` have already been removed in this PR.)

Reply:
You compose data blobs of different schema versions by promoting/converting the old one to the new version (or demoting the new one to the old version if it can be encoded in the old version, depending on requirements) and then you have a single version again.

But we aren't set up to do that and it's probably not worthwhile to try...